### ISSUE TITLE
More details

### ISSUE THREAD

**Issue by marvis** [Kaixhin/FCN-semantic-segmentation#2]
Hi Kaixhin,

I just want to try your demo. But I am new to FCN model. Can you give more details on how to process data, and how to run the demo successfully.

I met the same problem on pytorch that the train accuracy is low when train object detection. And I solved the problem with a few tricks. I am very interest in your code. And will be happy figure out your problem.

**Comment by Kaixhin** [comment]
I've added more details in the readme for everyone to see. Another minor detail that I haven't included is that the data is [normalised](https://github.com/Kaixhin/FCN-semantic-segmentation/blob/master/main.py#L78-L81) according to the [ImageNet preprocessing procedure used for PyTorch's pretrained models](https://github.com/pytorch/examples/blob/master/imagenet/main.py#L100-L101). Let me know if you have any more questions!

**Comment by marvis** [comment]
The problem in pytorch training is that all parameters have the same global weight decay including bn and bias. I find it a big problem to get good training result. You can  try to set bn and bias params' weight decay to zero according to the following code.  Maybe helps for you.

params_dict = dict(model.named_parameters())
params = []
for key, value in params_dict.items():
    if key.find('.bn') >= 0 or key.find('.bias') >= 0:
        params += [{'params': [value], 'weight_decay': 0.0}]
    else:
        params += [{'params': [value], 'weight_decay': decay*batch_size}]
optimizer = optim.SGD(model.parameters(), lr=learning_rate/batch_size, momentum=momentum, dampening=0, weight_decay=decay*batch_size)

**Comment by Kaixhin** [comment]
I've tried this now, plus a double learning rate for biases as I've seen in other places/been told by someone, but not seeing a change in results. Still seeing ~40 IoU with RMSprop, and nothing much at all with SGD+momentum.
